crates/bevy_ui/src/texture_slice.rs

```diff
@@ -1,213 +0,0 @@
-// This module is mostly copied and pasted from `bevy_sprite::texture_slice`
-//
-// A more centralized solution should be investigated in the future
-
-use bevy_asset::{AssetEvent, Assets};
-use bevy_ecs::prelude::*;
-use bevy_math::{Rect, Vec2};
-use bevy_render::texture::Image;
-use bevy_sprite::{ImageScaleMode, TextureAtlas, TextureAtlasLayout, TextureSlice};
-use bevy_transform::prelude::*;
-use bevy_utils::HashSet;
-
-use crate::{CalculatedClip, ExtractedUiNode, Node, NodeType, UiImage};
-
-/// Component storing texture slices for image nodes entities with a tiled or sliced  [`ImageScaleMode`]
-///
-/// This component is automatically inserted and updated
-#[derive(Debug, Clone, Component)]
-pub struct ComputedTextureSlices {
-    slices: Vec<TextureSlice>,
-}
-
-impl ComputedTextureSlices {
-    /// Computes [`ExtractedUiNode`] iterator from the sprite slices
-    ///
-    /// # Arguments
-    ///
-    /// * `transform` - the sprite entity global transform
-    /// * `original_entity` - the sprite entity
-    /// * `sprite` - The sprite component
-    /// * `handle` - The sprite texture handle
-    #[must_use]
-    pub(crate) fn extract_ui_nodes<'a>(
-        &'a self,
-        transform: &'a GlobalTransform,
-        node: &'a Node,
-        image: &'a UiImage,
-        clip: Option<&'a CalculatedClip>,
-        camera_entity: Entity,
-    ) -> impl ExactSizeIterator<Item = ExtractedUiNode> + 'a {
-        let mut flip = Vec2::new(1.0, -1.0);
-        let [mut flip_x, mut flip_y] = [false; 2];
-        if image.flip_x {
-            flip.x *= -1.0;
-            flip_x = true;
-        }
-        if image.flip_y {
-            flip.y *= -1.0;
-            flip_y = true;
-        }
```

Comment:
@ickshonpe thanks for the PR, using a shader is much better than CPU bound logic.
I don't see an equivalent to this flip image logic anywhere in the new code, so it might be missing from the migration

Reply:
Hi felix, yes you're right, I took image flipping out temporarily to make things easier and meant to add it back in at the end but completely forgot about it :sweat:. I think it must be really rare to combine nine slicing with flipping though so it's probably not too critical and shouldn't be difficult to add support in a followup pr.